src/wasm/wasm-type.cpp

```diff
@@ -766,6 +769,11 @@ TypeID Store<Info>::doInsert(std::unique_ptr<Info>&& info) {
   return id;
 }
 
+// For nominal mode, explicitly track all the subtype relations. Each HeapType
+// can have only one immediate supertype. TODO:
+std::unordered_map<HeapType, HeapType> nominalSuperTypes;
+std::mutex nominalSuperTypesLock;
```

Comment:
why is a separate lock needed for this? the class has a lock already, so i assume there is a specific design in mind here? comment would be good.

Reply:
This lock isn't part of a `Store`. It's a standalone lock that protects the single global `nominalSuperTypes`.